### ISSUE TITLE
Need warning in docs about bumping master branch during deployment to gh pages

### ISSUE THREAD

**Issue by Ildarik** [gatsbyjs/gatsby#24479]
Recently I faced same issue as described [here](https://github.com/gatsbyjs/gatsby/issues/13120)

Maybe we need some sort of warning or tip in [docs](https://www.gatsbyjs.org/docs/how-gatsby-works-with-github-pages/#deploying-to-a-github-pages-subdomain-at-githubio) about that issue?

**Comment by LekoArts** [comment]
Hi, thanks for the issue!

What issue did you face? The issue you linked is working as intended and the docs explain this usecase.

**Comment by Ildarik** [comment]
@LekoArts added description of the problem

**Comment by LekoArts** [comment]
Reading your updated comment I feel like you should follow these instructions, right?
https://www.gatsbyjs.org/docs/how-gatsby-works-with-github-pages/#deploying-to-a-path-on-github-pages

Because with this script you'll always deploy to another branch than `master`

**Comment by Ildarik** [comment]
No, correct link is https://www.gatsbyjs.org/docs/how-gatsby-works-with-github-pages/#deploying-to-a-github-pages-subdomain-at-githubio

GH pages forces to deploy to master branch for user\organization pages.

**Comment by LekoArts** [comment]
Ah, I see!

Documenting those two options (https://github.com/gatsbyjs/gatsby/issues/13120#issuecomment-479932834) sounds reasonable. Would you be up for opening a PR adding this? :)

**Comment by Ildarik** [comment]
Sure! 
I'll do it today a little bit later! Thanks!)
